plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightClientManager.java

```diff
@@ -184,13 +200,19 @@ private DiscoveryNode getNodeFromClusterState(String nodeId) {
      */
     @Override
     public void close() throws Exception {
-        for (FlightClient flightClient : flightClients.values()) {
-            flightClient.close();
+        if (closed) {
+            return;
+        }
+        closed = true;
+        for (ClientHolder clientHolder : flightClients.values()) {
+            clientHolder.flightClient.close();
         }
         flightClients.clear();
         grpcExecutor.shutdown();
         grpcExecutor.awaitTermination(5, TimeUnit.SECONDS);
```

Comment:
In worst case, let's do best effort shutdown:
```suggestion
        if (grpcExecutor.awaitTermination(5, TimeUnit.SECONDS) == false) {
            grpcExecutor.shutdownNow();
        }
```